### ISSUE TITLE
Proposal for using mirror image for Chinese developer with Alibaba Cloud Registry Service

### ISSUE THREAD

**Issue by denverdino** [kubernetes/minikube#12535]
**Steps to reproduce the issue:** 

There are some fixes for making Chinese developer to use Minikube with the mirrored images from Alibaba Cloud Registry Service. And I hope we can fix it more systematically. and I create one prototype for that.

1. Firstly, define the mirror mapping from the original repo to the mirrored repo, e.g. https://github.com/AliyunContainerService/minikube/blob/aliyun-v1.23.1/deploy/addons/aliyun_mirror.json

2. When enable one specific addon, it will check if the mirror will be used. If yes, it will generate the custom image and custom repo for override.

https://github.com/AliyunContainerService/minikube/blob/aliyun-v1.23.1/pkg/minikube/assets/aliyun_mirror.go

Any comment or suggestion for that?

BTW, I built some simple tool to sync the minikube images to Alibaba Cloud Registry Service "registry.cn-hangzhou.aliyuncs.com/google_containers" everyday. In case anything missed, pls open one issue for me, and I will fix it soon.  

Thanks

**Comment by medyagh** [comment]
hi @denverdino thank you for reaching out, who funds the  Alibaba Cloud Registry  account you are reffering to?
would it cost a lot to the owner of the project ?

**Comment by medyagh** [comment]
we been given an alibaba account for minikube but we have not set it up but I believe the funding for it is limitted and might not handle a lot of traffic

**Comment by denverdino** [comment]
> hi @denverdino thank you for reaching out, who funds the Alibaba Cloud Registry account you are reffering to?
> would it cost a lot to the owner of the project ?
> 
> btw I am curious, have u seen this PR ? #12411
> does that help with anything?

Yep, I found the previous sync rule from "docker.io/coredns/coredns" is incorrect. So I fix that by adding sync rule   

```
"k8s.gcr.io/coredns/coredns": "registry.cn-hangzhou.aliyuncs.com/google_containers/coredns",
```

And PR #12411 is not required anymore.

Thanks

**Comment by denverdino** [comment]
@medyagh  

My team was helping to fund such repo. And I am leading the development for Alibaba Cloud Container Service, and we are happy to making the life of Kubernetes developer in China more easier. I will talk to you in slack

**Comment by denverdino** [comment]
Supported with #12578 and merged in minikube v1.24.0 !
